pkg/auth/dependency/sso/sso.go

```diff
@@ -38,7 +38,8 @@ type State struct {
 	LoginState
 	LinkState
 	OAuthAuthorizationCodeFlowState
-	Nonce string `json:"nonce,omitempty"`
+	Nonce  string `json:"nonce,omitempty"`
+	APIKey string `json:"api_key"`
```

Comment:
I think it's better to store just the client ID in OAuth state, since the state is already authenticated using JWT. API key can be considered as a sensitive information so we should minimize where it would be stored.

Reply:
I had discussed this with Carmen and we thought API Key is public and Client ID is private (at least it only appears in tenant configuration) so I included API key instead.

Reply:
For me, I don't think API key is public, otherwise we don't need this concept: API key is supposed to be accessible only to authorized clients (e.g. if somebody somehow get the server endpoint, they can't call API without also finding a valid API key from some client). Therefore API key should be considered a private information.

Personally I consider API key a more private (and sensitive) information than client ID: it is kind of like an access token for API.

Reply:
@carmenlau Any thoughts?

Reply:
Right now Client ID does not have any special meanings. We do not accept Client ID in any endpoints. So even it is known by a third party, they should not be able to do anything with it. As long as we can uphold this assumption. We can change (should change) to Client ID.

Reply:
In my opinion ID should not be a kind of sensitive info, so exposing client ID should not have major security implication.